File: 1.js
function bubbleSort(arr) {
    const n = arr.length;
    // Variable to track if any swap occurred in a pass
    let swapped;

    for (let i = 0; i < n; i++) {
        // If no swaps occur in a pass, array is sorted
        if (!swapped) {
            break;
        }
        
        // Set swapped to false at the beginning of each pass
        swapped = false;
        
        // Last i elements are already in place, so no need to check them
        for (let j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                // Swap if the current element is greater than the next one
                [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
                // Set swapped to true indicating a swap occurred
                swapped = true;
            }
        }
    }
}

// Example usage:
const arr = [64, 25, 12, 22, 11];
bubbleSort(arr);
console.log("Sorted array:");
console.log(arr);

File: 1.py
def bubble_sort(arr):
    n = len(arr)
    
    swapped = True

    for i in range(n):
        
        if not swapped:
            break
        swapped = False
        for j in range(0, n-i-1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
                swapped = True

arr = [64, 25, 12, 22, 11]
bubble_sort(arr)
print("Sorted array:")
print(arr)
